File: 알고리즘/CodingTest/src/Programmers/level1/Hashard.java
package Programmers.level1;

// https://programmers.co.kr/learn/courses/30/lessons/12947

public class Hashard {
    public boolean solution(int x) {
        boolean answer = false;
        int sum = 0;
        int mod = x;

        while(x > 0) {
           sum += x % 10; // 한 자리 수씩 더하기
           x /= 10; // 자릿수 줄이기
        }

        if(mod % sum == 0) {
            answer = true;
        }

        return answer;
    }
}

File: Algorithm/Algorithm_py/programmers1/Hashard.py
# https://programmers.co.kr/learn/courses/30/lessons/12947?language=python3
def solution(x):
  answer = True
  sum = 0
  x_list = list(map(int, str(x)))
  
  for i in x_list:
    sum += i
  
  if x % sum != 0:
    answer = False
  
  return answer